modules/effects/src/provide_effects.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  ENVIRONMENT_INITIALIZER,
+  inject,
+  InjectFlags,
+  Type,
+} from '@angular/core';
+import {
+  EnvironmentProviders,
+  FEATURE_STATE_PROVIDER,
+  ROOT_STORE_PROVIDER,
+  Store,
+} from '@ngrx/store';
+import { EffectsRunner } from './effects_runner';
+import { EffectSources } from './effect_sources';
+import { rootEffectsInit as effectsInit } from './effects_actions';
+
+/**
+ * Runs provided effects. Can be called at the root or feature level.
+ * Unlike the `EffectsModule.forRoot` method, this function does not
+ * need to be called at the root level if there are no root effects.
+ *
+ * @usageNotes
+ *
+ * ### Providing root effects
+ *
+ * ```ts
+ * bootstrapApplication(AppComponent, {
+ *   providers: [provideEffects([RouterEffects])],
+ * });
+ * ```
+ *
+ * ### Providing feature effects
+ *
+ * ```ts
+ * const booksRoutes: Route[] = [
+ *   {
+ *     path: '',
+ *     providers: [provideEffects([BooksApiEffects])],
+ *     children: [
+ *       { path: '', component: BookListComponent },
+ *       { path: ':id', component: BookDetailsComponent },
+ *     ],
+ *   },
+ * ];
```

Comment:
I think "unlike" could be confusing, what about leaving it out?
On the same note, we could also reword the root and feature effects to make it clear that there is no distinction, and thus that you can use it in the various levels.

Does that make sense? 

```suggestion
 * Runs the provided effects. 
 * Can be called at the root and feature levels.
 *
 * @usageNotes
 *
 * ### Providing effects at the root level
 *
 * ```ts
 * bootstrapApplication(AppComponent, {
 *   providers: [provideEffects([RouterEffects])],
 * });
 * ```
 *
 * ### Providing effects at the feature level
 *
 * ```ts
 * const booksRoutes: Route[] = [
 *   {
 *     path: '',
 *     providers: [provideEffects([BooksApiEffects])],
 *     children: [
 *       { path: '', component: BookListComponent },
 *       { path: ':id', component: BookDetailsComponent },
 *     ],
 *   },
 * ];
```